src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -99,31 +110,49 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
         }
 
         if (request.getWorkflowId() == null) {
-            // Create new global context and state index entries
-            flowFrameworkIndicesHandler.putTemplateToGlobalContext(templateWithUser, ActionListener.wrap(globalContextResponse -> {
-                flowFrameworkIndicesHandler.putInitialStateToWorkflowState(
-                    globalContextResponse.getId(),
-                    user,
-                    ActionListener.wrap(stateResponse -> {
-                        logger.info("create state workflow doc");
-                        listener.onResponse(new WorkflowResponse(globalContextResponse.getId()));
+            // Throttle incoming requests
+            onSearchGlobalContext(request.getRequestTimeout(), request.getMaxWorkflows(), ActionListener.wrap(max -> {
```

Comment:
Perhaps lets rename this method

Reply:
Sure. Any suggestion?

Reply:
Changed it to `checkMaxWorkflows`.

Reply:
~~superAwesomeMethodToCheckWorkflowsAndThrottleImcomingRequests~~

maybe `shouldThrottleRequests()` following some naming conventions [here](https://github.com/opensearch-project/anomaly-detection/blob/4c6ba48bf9fb9234ad8bc0b9193f3c68409acfb9/src/main/java/org/opensearch/timeseries/indices/IndexManagement.java#L688)

Reply:
`checkMaxWorkflows` works too :)